Set SYSTEMD_IGNORE_CHROOT=yes

rayandas opened the pull request:
Related issue: https://github.com/rancher/rancher/issues/41601

As systemd installed in the latest hyperkube-base, when the user is trying to start a systemd service, it's running within a chroot environment, hence the "start" command is being ignored. Refer to the below error:

```
# systemctl start rpc-statd.service
Running in chroot, ignoring command ‘start’
```

This PR adds an env variable SYSTEMD_IGNORE_CHROOT and set it to yes so that it ignores the chroot environment.

superseb reviewed (changes requested):
Can you include how it was tested and/or if the linked issue was reproduced and solved?

rayandas commented:
@superseb I couldn't reproduce the issue but after going through the comments on the issue it's certain that `if systemctl start rpc-statd.service` is giving this error `Running in chroot, ignoring command ‘start’`, which means it's running within a chroot environment. So any systemctl command will give the same chroot error. And that is why the script in `/usr/sbin/start-statd` is failing. As we can't modify the script, I added `ENV SYSTEMD_IGNORE_CHROOT=yes` to the Dockerfile. So it instructs systemd to ignore the chroot environment when running systemd commands. 

So I built hyperkube-base using this env and updated hyperkube and KDM, then provisioned an RKE1 cluster, ssh-ed into the node and exec into kubectl container, ran a systemctl command, it's running successfully and not giving the chroot error. 

So now the script in `/usr/sbin/start-statd` will runn successfully and if systemd is present, it'll not stop rpc.statd or its dependencies. And if systemd is not present then it'll run the rpc.statd manually using `exec rpc.statd --no-notify` as mentioned on the script.

jiaqiluo commented on Dockerfile:
```diff
@@ -1,6 +1,7 @@
 FROM ubuntu:22.04
 
 ARG ARCH=amd64
+ENV SYSTEMD_IGNORE_CHROOT=yes
```
According to the [systemd docs](https://systemd.io/ENVIRONMENT/), the proper value to use is `1`, can you change it to 1 and also validate it works?

rayandas replied:
Sure

jiaqiluo reviewed (approved):
Hi, @rayandas. Can you confirm if the fix works after changing the value to `1`?  
I will approve the PR once I hear your confirmation.

chiukapoor commented:
Hello @jiaqiluo @superseb, below are the steps followed to reproduce and test the linked issue.

- Created an EFS store on AWS.
- Deployed a three-node Kubernetes cluster using Rancher.
- Deployed Mongodb community operator on the above cluster.
- Created Persistent Volumes using the NFS store provided by EFS and the Local hostPath store for logs.
- Installed `nfs-common` package on all the Kubernetes cluster nodes
```
apt-get install -y nfs-common
```
- Created mongoDBCommunity object with below yaml
```
apiVersion: mongodbcommunity.mongodb.com/v1
kind: MongoDBCommunity
metadata:
  name: example-mongodb
  namespace: default
spec:
  members: 3
  type: ReplicaSet
  version: "6.0.5"
  security:
    authentication:
      modes: ["SCRAM"]
  users:
    - name: my-user
      db: admin
      passwordSecretRef:
        name: my-user-password
      roles:
        - name: clusterAdmin
          db: admin
        - name: userAdminAnyDatabase
          db: admin
      scramCredentialsSecretName: my-scram
  additionalMongodConfig:
    storage.wiredTiger.engineConfig.journalCompressor: zlib
  statefulSet:
    spec:
        # Name for the service object created by the operator
      serviceName: example-mongodb
        # Specifies a size for the data volume different from the default 10Gi
      volumeClaimTemplates:
        - metadata:
            name: data-volume
          spec:
            accessModes: 
              - ReadWriteOnce
            resources:
              requests:
                storage: 10Gi
      template:
        # Adds a custom volume to the pods
        spec:
          #  Hostpath volumes are owned by root
          #  but MongoDB containers run as non root
          #  so we use an init container to change the owner of
          #  the directory (init containers run as root)
          initContainers:
          - command:
              - chown
              - -R
              - "2000"
              - /data
            image: busybox
            volumeMounts:
            - mountPath: /data
              name: data-volume
            securityContext:
              runAsNonRoot: false
              runAsUser: 0
              runAsGroup: 0
            name: change-dir-permissions
          volumes:
            - name: custom-volume
              emptyDir: {}
          containers:
          - name: mongodb-agent
            volumeMounts:
              - name: custom-volume
                mountPath: /my-custom-version/mount-path
---
apiVersion: v1
kind: Secret
metadata:
  namespace: default
  name: my-user-password
type: Opaque
stringData:
  password: test123
```
- Pod creation failed with
```
{"t":{"$date":"2023-11-14T10:13:38.939+00:00"},"s":"E",  "c":"CONTROL",  "id":20557,   "ctx":"initandlisten","msg":"DBException in initAndListen, terminating","attr":{"error":"DBPathInUse: Unable to lock the lock file: /data/mongod.lock (Resource temporarily unavailable). Another mongod instance is already running on the /data directory"}}
```
- MongoDB resource deployed successfully while testing with the hyperkube image created after the fix.

rayandas commented:
Closing on behalf of https://github.com/rancher/hyperkube-base/pull/20
